services/base-http.js

```diff
@@ -0,0 +1,22 @@
+'use strict'
+
+// See available emoji at http://emoji.muan.co/
+const emojic = require('emojic')
+const { checkErrorResponse } = require('../lib/error-helper')
+const BaseService = require('./base')
+const trace = require('./trace')
+
+class BaseHTTPService extends BaseService {
```

Comment:
We already have three layers in this inheritance tree. It's already one more than I'd like, and I'd like to avoid adding a fourth. Inherited behaviors are difficult to understand, and on account of all the implicit dependencies, even more difficult to change.

Could this code be moved into BaseService? While there are a couple services which don't make a request, this is a tiny fraction and I it's not worth making major design decisions around them. I'd advocate for calling it `_request` since HTTP is the only kind we make. Right now there are no services using `_sendAndCacheRequest` directly. If any others did need to, this behavior would be useful.

Reply:
You can choose how you'd like to move forward:

1. Copy this implementation into the FDroid service. Reset the related changes in the base classes. Open a new issue for refactoring this and adding tests. This approach will result in a faster merge.
2. Continue the discussion here about moving it into BaseService and adding tests.

Reply:
Moving this function to the base class _may_ help for a piece of work that I'm currently doing. I'm happy to take care of this review comment if that helps. We can merge this bit of the PR as is and I'm happy to do the refactoring afterwards. @niccokunzmann let me know what you think. 😉